packages/smart-contracts/src/contracts/BatchPayments.sol

```diff
@@ -96,8 +89,10 @@ contract BatchPayments is Ownable, ReentrancyGuard {
         _feeAddress.transfer(sumBatchFeeAmount);
 
         // Transfer the remaining ethers to the sender
-        (bool sendBackSuccess, ) = payable(msg.sender).call{ value: toReturn }('');
-        require(sendBackSuccess, 'Could not send remaining funds to the payer');
+        if (remainingAmount > 0) {
+            (bool sendBackSuccess, ) = payable(msg.sender).call{ value: remainingAmount }('');
+            require(sendBackSuccess, 'Could not send remaining funds to the payer');
```

Comment:
Shouldn't we send back 100% of the balance?

Reply:
Done